src/Browser.php

```diff
@@ -725,6 +728,72 @@ public function withResponseBuffer($maximumSize)
         ));
     }
 
+    /**
+     * Add a request header for all following requests.
+     *
+     * ```php
+     * $browser = $browser->withHeader('User-Agent', 'ACME');
+     *
+     * $browser->get($url)->then(function (Psr\Http\Message\ResponseInterface $response) {
+     * //will contain the header User-Agent with the value ACME
+     * var_dump($response);
+     * }, function (Exception $e) {
+     * echo 'Error: ' . $e->getMessage() . PHP_EOL;
+     * });
+     * ```
+     *
+     * Note that the new header will overwrite any headers previously set with
+     * the same name (case-insensitive). Following requests will use these headers
+     * by default unless they are explicitly set for any requests.
+     *
+     * @param string $header
+     * @param string $value
+     * @return Browser
+     */
+    public function withHeader($header, $value)
+    {
+        $browser = $this->withoutHeader($header);
+        $browser->defaultHeaders[$header] = $value;
+
+        return $browser;
+    }
+
+    /**
+     * Remove any default request headers previously set via
+     * the [`withHeader()` method](#withheader).
+     *
+     * ```php
+     * $browser = $browser->withoutHeader('User-Agent');
+     *
+     * $browser->get($url)->then(function (Psr\Http\Message\ResponseInterface $response) {
+     * //won't contain the User-Agent header
+     * var_dump($response);
+     * }, function (Exception $e) {
+     * echo 'Error: ' . $e->getMessage() . PHP_EOL;
+     * });
+     * ```
```

Comment:
Nit, to match the example in `README.md`:
```suggestion
     * ```php
     * $browser = $browser->withoutHeader('User-Agent');
     *
     * $browser->get($url)->then(function (Psr\Http\Message\ResponseInterface $response) {
     *     //won't contain the User-Agent header
     *     var_dump($response);
     * }, function (Exception $e) {
     *     echo 'Error: ' . $e->getMessage() . PHP_EOL;
     * });
     * ```
```